### PR TITLE
Fix taxonomy current path

### PR DESCRIPTION
**Pull request by bemyak**
Fixes https://github.com/getzola/zola/issues/1878

---

### Sanity check:

* [X] Have you checked to ensure there aren't other open [Pull Requests](https://github.com/getzola/zola/pulls) for the same update/change?
* [X] Are you doing the PR on the `next` branch?

### CODE REVIEW

File: components/templates/src/global_fns/files.rs [hunk 1]
```diff
@@ -104,7 +104,7 @@ impl TeraFn for GetUrl {
             // anything else
             let mut segments = vec![];
 
-            if lang != self.config.default_language {
+            if lang != self.config.default_language && !path.starts_with(&format!("/{}", lang)) {
```
Comment (Keats) [inline components/templates/src/global_fns/files.rs]: you can do that without allocating by slicing I believe?

File: components/templates/src/global_fns/files.rs [hunk 2]
```diff
@@ -399,6 +399,29 @@ title = "A title"
         );
     }
 
+    #[test]
+    fn does_not_duplicate_lang() {
```
Comment (Keats) [inline components/templates/src/global_fns/files.rs:403]: what does it mean to duplicate lang?

File: components/content/src/taxonomies.rs [hunk 3]
```diff
@@ -204,8 +203,8 @@ impl Taxonomy {
         context.insert("terms", &terms);
         context.insert("lang", &self.lang);
         context.insert("taxonomy", &self.kind);
-        context.insert("current_url", &config.make_permalink(&self.kind.name));
-        context.insert("current_path", &format!("/{}/", self.kind.name));
+        context.insert("current_url", &self.permalink);
+        context.insert("current_path", &self.path);
```
Comment (Keats) [inline components/content/src/taxonomies.rs:207]: so we already had them eh

**Review by Keats — CHANGES_REQUESTED** [review]

File: components/templates/src/global_fns/files.rs (hunk 1 above)
Comment (bemyak) [reply components/templates/src/global_fns/files.rs]: Can you elaborate please?

File: components/templates/src/global_fns/files.rs (hunk 2 above)
Comment (bemyak) [reply components/templates/src/global_fns/files.rs:403]: It's described in https://github.com/getzola/zola/issues/1878.

TL;DR: currenly `get_url(path='/lang/something', lang='lang')` will return a URL with duplicated `lang` part: `/lang//lang/something`

File: components/content/src/taxonomies.rs (hunk 3 above)
Comment (bemyak) [reply components/content/src/taxonomies.rs:207]: Yes, we indeed had both `permalink` and `path` already computed in the `new` function, but `path` was discarded and `permalink` wasn't used. I've made it to match other similar structs e.g., `Path`

File: components/templates/src/global_fns/files.rs (hunk 1 above)
Comment (Keats) [reply components/templates/src/global_fns/files.rs]: Something like `path[1..].starts_with(lang)`. Haven't tried it so it might not work

Comment (bemyak) [reply components/templates/src/global_fns/files.rs]: Indeed, this works! Thanks for the handy trick :)

I've amended the last commit.

**Review by Keats — APPROVED** [review]
